src/HTTP1/HTTP1.ts

```diff
@@ -81,16 +86,29 @@ Host: ${request.url.hostname}\r
                     if (rnrn !== -1) {
                         this._parseHeaders(rnrn);
                         this.headersFinished = true;
-                        const remaining = this.headerBuffer.byteLength - (rnrn + 4);
+
+                        let remaining = this.headerBuffer.byteLength - (rnrn + 4);
+
+                        const hOffset = this.headerBuffer.byteLength - remaining;
+                        if (!this.chunkyParser && !this.contentLength) {
```

Comment:
Technically this does not cover a case.  It was lost in the merge.  there should be an else statement that checks to see if contentLength is less than the header buffer and should unread it.